src/mercury_engine_data_structures/formats/bmsad.py

```diff
@@ -294,23 +315,35 @@ def SRDependencies():
     model_name=StrId,
 
     unk_1=Int8ul,
-    unk_2=Float[9],
+    unk_2a=Float,
+    unk_2b=Float,
+    unk_2c=Float,
+    unk_2d=Float,
+    unk_2e=Float,
+    unk_2f=CVector3D,
+    unk_2g=Float,
     unk_3=Int8ul,
     unk_4=Int32ul,
-    other_magic=Const(0xFFFFFFFF, Hex(Int32ul)),
-    unk_5=Int16ul,
-    unk_6=StrId[2],
-    unk_7=Int8ul,
+    other_magic=Int32sl,
+    unk_5=Flag,
+    unk_5b=Flag,
+    unk_6=Flag,
+    category=StrId,
+    unk_7=Flag,
     sub_actors=make_vector(StrId),
     unk_8=Int32ul,
 
-    # count=Int32ul,
-    # component=(StrId >> SR_Component)[4],
     components=make_dict(SR_Component),
 
-    binaries=make_vector(StrId),
+    unk_9=Int32ul,
+    unk_9a=construct.If(lambda this: this.unk_9 == 2, construct.Bytes(9)),
+    unk_9b=construct.If(lambda this: this.unk_9 == 0, construct.Bytes(15)),
+    unk_10=StrId,
+    unk_11=Int32ul,
+
+    action_sets=make_vector(BMSAS_SR),
+    sound_fx=construct.Optional(make_vector(StrId >> Byte)),
```

Comment:
Is there really nothing you can use to identify that this field is present? Optional sucks as it silently ignores errors.

Reply:
i can probably do better. for whatever reason if it would be length zero it just doesn't include it at all which is annoying

Reply:
Can you make it a `Select(vector >> Terminated, Terminated)`?

Though I guess if the Optional fails it'll then fail matching at the Terminated already 🤔  And the select would still cause a failure to parse the strings to be silently ignored ugh